### PR TITLE
policy: Add Regal for linting Rego

### PR DESCRIPTION
**Pull request by anderseknert**
This PR introduces [Regal](https://github.com/styrainc/regal) for linting the Rego files of the project. The Rego found here is generally in a good shape, so only a few rules have been either ignored or fixed. The few rules fixed include:
* [rule-shadows-builtin](https://docs.styra.com/regal/rules/bugs/rule-shadows-builtin) as there is already an `is_null` function in the standard lib
* [custom-has-key-construct](https://docs.styra.com/regal/rules/idiomatic/custom-has-key-construct) as this is handled by `object.keys` and `in` in modern Rego
* [use-assignment-operator](https://docs.styra.com/regal/rules/style/use-assignment-operator) as `:=` is preferred over `=` for rule assignment, and using this does not change the semantics of the code in any way.

Included is also a new job for checking/linting Rego code as part of CI.

### CODE REVIEW

File: .github/workflows/build_and_test.yaml [hunk 1]
```diff
@@ -27,3 +27,16 @@ jobs:
         run: go test -count=1 -shuffle=on -timeout=10m -race $(go list ./... | grep -vE '^github.com/Legit-Labs/legitify/e2e')
       - name: Vet
         run: go vet -v ./...
+  check_rego:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: open-policy-agent/setup-opa@v2
```
Comment (gal-legit) [inline .github/workflows/build_and_test.yaml:34]: Please pin the commit for this action

File: .github/workflows/build_and_test.yaml [hunk 2]
```diff
@@ -27,3 +27,16 @@ jobs:
...
+        with:
+          version: v0.55.0
+      - uses: styrainc/setup-regal@v0.2.0
```
Comment (gal-legit) [inline .github/workflows/build_and_test.yaml:37]: Please pin the commit for this action

**Comment by gal-legit** [comment]
Hey @anderseknert,
thank you so much for this awesome contribution!
I commented on a couple of new actions that need no be pinned down to a specific commit version to avoid a risk. Please just sort it out and I'll approve and merge.

**Comment by anderseknert** [comment]
Thanks @gal-legit 😃 I've done that now 👍

**Review by royb-legit — APPROVED** [review]

**Comment by royb-legit** [comment]
Thank you for your contribution, @anderseknert!

**Pull request merged**